### DISCUSSION TITLE
Is it worth getting rid of the lists of dependencies in `package.json`?

### DISCUSSION THREAD

**Discussion by cawa-93** [npm/feedback#321] [CLI]
It seems that `package-lock.json` contains all the necessary information about dependencies and environments. In fact, it duplicates the `*dependencies` and `engines` fields from ` package.json`. So why not get rid of these fields in `package.json`?

In this case, we will have two files:
- `package-lock.json` - Contains all information about dependencies and environments. Commands such as `npm install` and` npm update` will only update this file.
- `package.json` - Containing all meta-information about the package (project), name, scripts, configuration and so on. Without a list of dependencies, this file will be smaller and easier to understand.

**Comment by ljharb** [comment]
package-lock is not published, so a published package must have `package.json`. Additionally, the lockfile can (and occasionally should) be regenerated from scratch - the source of truth is package.json.

**Reply by MylesBorins** [reply]
For the above mentioned reasons and the fact that such a change would be breaking and non backwards-compatible I cannot see a future where we remove dependencies from the package.json

if you would like to make a formal proposal about this to the CLI team you can open an issue over at https://github.com/npm/rfcs, but I am not bullish that it would be accepted.
